workflows/prognostic_c48_run/runtime/emulator.py

```diff
@@ -145,44 +140,3 @@ def step() -> Diagnostics:
         step.__name__ = func.__name__
 
         return step
-
-
-def _update_state_with_emulator(
-    state: MutableMapping[Hashable, xr.DataArray],
-    src: Mapping[Hashable, xr.DataArray],
-    compute_mask: Callable[[Hashable, xr.DataArray], xr.DataArray],
-) -> None:
-    """
-    Args:
-        state: the mutable state object
-        src: updates to put into state
-        compute_mask: a function returning a mask. Where this mask is True, the
-            original state array will be used.
-
-    """
-    for key in src:
-        arr = state[key]
-        mask = compute_mask(key, arr)
-        state[key] = arr.where(mask, src[key].variable)
-
-
-@dataclasses.dataclass
-class compute_mask:
-    ignore_humidity_below: Optional[int] = None
-
-    def __call__(self, name: Hashable, arr: xr.DataArray) -> xr.DataArray:
-        if name == SPHUM:
-            if self.ignore_humidity_below is not None:
-                return arr.z > self.ignore_humidity_below
-            else:
-                return xr.DataArray(False)
-        else:
-            return xr.DataArray(True)
-
-
-def update_state_with_emulator(
```

Comment:
I'd propose we leave `update_state_with_emulator` function in this module and within that, optionally apply the mask when specified